docs/COMPLEX_COMPOSITION.md

```diff
@@ -16,7 +16,166 @@ rule <rule name> [when <condition>] {
 
 Where Guard rule is an umbrella term for clause, query block, `when` block or named rule block.
 
-1. Named rule blocks allow for re-usability, improved composition, and reduced verbosity and repetition. This document will focus on demonstrating these features of named rule blocks in-depth.
+Query blocks and named rule blocks allow for re-usability, improved composition, and reduced verbosity and repetition. This document will focus on demonstrating these features in-depth.
+
+## Clause evaluation
+
+Each clause is an independent statement whose result is the evaluation on its entire query result based on the clause's scope. This is best explained with an example,
```

Comment:
This is a very dense and difficult sentence to follow. I am unable to understand what you are trying to convey here. I  would rather start with an example set of clauses and then break it down for the reader.